interoperability-layer-openhim/swarm.sh

```diff
@@ -1,171 +1,151 @@
-    config::timeout_check "${start_time}" "mongo set to start"
-    sleep 1
-
-    running_instance_count=0
-    for i in $(docker service ls -f name=instant_mongo --format "{{.Replicas}}"); do
-      if [[ "${i}" == "1/1" ]]; then
-        running_instance_count=$((running_instance_count + 1))
-      fi
-    done
-  done
+# shellcheck disable=SC1091
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/config-utils.sh"
+  source "${UTILS_PATH}/log.sh"
 }
 
-prepare_console_config() {
+function prepare_console_config() {
   # Set host in OpenHIM console config
   sed -i "s/localhost/${OPENHIM_CORE_MEDIATOR_HOSTNAME}/g; s/8080/${OPENHIM_MEDIATOR_API_PORT}/g" /instant/interoperability-layer-openhim/importer/volume/default.json
 }
 
-main() {
-  if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-    log info "Running Interoperability Layer OpenHIM package in Cluster node mode"
-    mongo_cluster_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose-mongo.cluster.yml"
-  else
-    log info "Running Interoperability Layer OpenHIM package in Single node mode"
-    mongo_cluster_compose_param=""
-  fi
+function initialize_package() {
+  local mongo_cluster_compose_filename=""
+  local mongo_dev_compose_filename=""
+  local openhim_dev_compose_filename=""
 
   if [[ "${MODE}" == "dev" ]]; then
-    log info "Running Interoperability Layer OpenHIM package in DEV mode"
-    local mongo_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose-mongo.dev.yml"
-    local openhim_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.dev.yml"
+    package::log info "Running package in DEV mode"
+    mongo_dev_compose_filename="docker-compose-mongo.dev.yml"
+    openhim_dev_compose_filename="docker-compose.dev.yml"
   else
-    log info "Running Interoperability Layer OpenHIM package in PROD mode"
-    local mongo_dev_compose_param=""
-    local openhim_dev_compose_param=""
+    package::log info "Running package in PROD mode"
   fi
 
-  if [[ "${ACTION}" == "init" ]]; then
-    config::set_config_digests "${COMPOSE_FILE_PATH}"/docker-compose.yml
-    config::set_config_digests "${COMPOSE_FILE_PATH}"/importer/docker-compose.config.yml
+  if [[ "${NODE_MODE}" == "cluster" ]]; then
+    mongo_cluster_compose_filename="docker-compose-mongo.cluster.yml"
+  fi
 
-    try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose-mongo.yml $mongo_cluster_compose_param $mongo_dev_compose_param instant" "Failed to deploy mongo"
+  (
+    docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose-mongo.yml" "$mongo_cluster_compose_filename" "$mongo_dev_compose_filename"
+    docker::deploy_sanity "${mongo_services[@]}"
 
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      try "${COMPOSE_FILE_PATH}/initiateReplicaSet.sh" "Fatal: Initate Mongo replica set failed."
+    if [[ "${NODE_MODE}" == "cluster" ]]; then
+      try "${COMPOSE_FILE_PATH}/initiate-replica-set.sh" throw "Fatal: Initiate Mongo replica set failed"
```

Comment:
Initiating the replicaset should be performed when the action = "init"

Reply:
done :)